container/go/cmd/create_image_config/createImageConfig.go

```diff
@@ -0,0 +1,413 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+////////////////////////////////////////////////
+// This package manipulates v2.2 image configuration metadata.
+// It writes out both a config file and a manifest for the v2.2 image.
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/pkg/errors"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	// "github.com/google/go-containerregistry/pkg/v1/mutate"
+)
+
+var (
+	baseConfig       = flag.String("baseConfig", "", "The parent image.")
+	baseManifest     = flag.String("baseManifest", "", "The parent image manifest.")
+	outputConfig     = flag.String("outputConfig", "", "The output file to generate.")
+	outputManifest   = flag.String("outputManifest", "", "The manifest output file to generate.")
+	creationTime     = flag.String("creationTime", "", "The creation timestamp. Acceptable formats: Integer or floating point seconds since Unix Epoch, RFC, 3339 date/time.")
+	user             = flag.String("user", "", "The username to run the commands under.")
+	workdir          = flag.String("workdir", "", "Set the working directory of the layer.")
+	nullEntryPoint   = flag.String("nullEntryPoint", "false", "If true, Entrypoint will be set to null.")
+	nullCmd          = flag.String("nullCmd", "false", "If true, Cmd will be set to null.")
+	operatingSystem  = flag.String("operatingSystem", "linux", "Operating system to create docker image for distro specified.")
+	labelsArray      arrayFlags
+	ports            arrayFlags
+	volumes          arrayFlags
+	entrypointPrefix arrayFlags
+	env              arrayFlags
+	command          arrayFlags
+	entrypoint       arrayFlags
+	layer            arrayFlags
+	stampInfoFile    arrayFlags
+	// empty file sha256 hex
+	b         = sha256.Sum256([]byte(""))
+	emptyFile = hex.EncodeToString(b[:])
+)
+
+// default architecture type based on legacy create_image_config.py
+const processorArchitecture = "amd64"
+
+// unix epoch 0, representation in 32 bits
+const defaultTimestamp = "1970-01-01T00:00:00Z"
+
+// arrayFlags is a type that can handle multiple flags of the same name.
+// the resulting data parsed in will be an array data type.
+type arrayFlags []string
+
+func (i *arrayFlags) String() string {
+	return fmt.Sprintf("%s", strings.Join(*i, ", "))
+}
+
+func (i *arrayFlags) Set(value string) error {
+	*i = append(*i, value)
+	return nil
+}
+
+// extractValue returns the contents of a file pointed to by value if it starts with a '@'.
+func extractValue(value string) (string, error) {
+	if strings.HasPrefix(value, "@") {
+		f, err := os.Open(value[1:])
+		if err != nil {
```

Comment:
This check is unnecessary. ioutil.Readall would return something similar if the file can't be opened. I suggest just removing this function as the only thing it's doing is reading the contents of a file and just calling this directly at the call site.